evennia/contrib/collab/commands.py

```diff
@@ -0,0 +1,817 @@
+                     "owner.{n")
+            return
+
+        set_owner(self.caller, target)
+        self.msg("{gOwner for %s(#%s) set to %s(#%s).{n"
+                 % (target.name, target.id, self.caller, self.caller.id))
+
+
+class CmdLink(MuxCommand):
+    """
+    Set the destination of an object. Most useful for changing the place where
+    exits will lead.
+
+    Usage: @link exit=destination
+    """
+    key = '@link'
+
+    def func(self):
+        if not self.lhs:
+            self.msg("{rUsage: @link thing=destination{n")
+        source = self.caller.search(self.lhs)
+        if not source:
+            return
+        if self.rhs:
+            destination = self.caller.search(self.rhs)
+            if not destination:
+                return
+        else:
+            destination = None
+        if not collab_check(self.caller, source, locks=['link']):
+            self.msg("{rYou don't have permission to link to '%s'.{n" % source)
+            return
+        if destination and not collab_check(self.caller, destination,
+                                            locks=['link_to']):
+            self.msg("{rYou don't have permission to link to '%s'."
+                     % destination.name)
+            return
+        source.destination = destination
+        if destination is None:
+            self.msg("{g'%s' unlinked." % source.name)
+        else:
+            self.msg("{g'%s' linked to '%s'."
+                     % (source.name, destination.name))
+
+
+class CmdSetHome(MuxCommand):
+    """
+    Set the home of an object.
+
+    Usage: @home object=home
+    """
+    key = '@home'
+
+    def func(self):
+        if not self.rhs:
+            self.msg("{rUsage: @home thing=home{n")
+        obj = self.caller.search(self.lhs)
+        if not obj:
+            return
+        if self.rhs:
+            home = self.caller.search(self.rhs)
+            if not home:
+                return
+        else:
+            home = None
+        if not collab_check(self.caller, obj, locks=['home']):
+            self.msg("{rYou don't have permission to set the home for '%s'.{n"
+                     % obj)
+            return
+        if home and not collab_check(self.caller, home, locks=['home_to']):
+            self.msg("{rYou don't have permission to set home to '%s'."
+                     % home.name)
+            return
+        obj.home = home
+        if home is None:
+            self.msg("{g'%s' home cleared. {yWarning, this object may get "
+                     "lost if its container is destroyed.{n" % obj.name)
+        else:
+            self.msg("{g'%s' home set to '%s'."
+                     % (obj.name, home.name))
+
+
+class CmdDesc(MuxCommand):
+    """
+    Describe an object. To give an object a one-line description, use:
+
+    @desc object=description
+
+    To enter a line editor for a longer description, use:
+
+    @desc/long object
```

Comment:
@Griatch I could go for either. `@desc/edit` is fine with me.

Perhaps a more interesting question while we're here, though, is concerning the line editor. Use of collab, by including this, causes the line editor to be a dependency. There's nothing necessarily wrong with this, but I wonder if the Line editor should become part of core instead of a contrib, and then be overridable by setting, and then this feature backported to the existing `@desc` command.